File: rust/client/src/components/shared/Btn/Btn.js
import React, {memo} from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import classes from './Btn.css';

Btn.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  isActive: PropTypes.bool,
  isDisabled: PropTypes.bool,
  onClick: PropTypes.func
};

function Btn({children, className, isActive, isDisabled, onClick}) {
  return (
    <button
      className={cn(classes.root, className, {[classes.active]: isActive})}
      disabled={isDisabled}
      onClick={onClick}
    >
      {children}
    </button>
  );
}

export default memo(Btn);

File: rust+yew/client/src/components/shared/btn/btn.rs
use yew::prelude::*;

#[derive(Properties, PartialEq, Clone)]
pub struct Props {
    #[prop_or_default]
    pub class: Classes,
    #[prop_or(false)]
    pub is_active: bool,
    #[prop_or(false)]
    pub is_disabled: bool,
    pub children: Children,
    pub on_click: Callback<MouseEvent>,
}

#[function_component(Btn)]
pub fn btn(props: &Props) -> Html {
    log::trace!("render Btn");
    let css = css_mod::get!("btn.css");
    html! {
        <button
            class={classes!(css["root"], props.is_active.then(|| css["active"]), props.class.clone())}
            disabled={props.is_disabled}
            onclick={props.on_click.clone()}
        >
            {for props.children.iter()}
        </button>
    }
}
